prow/config/agent.go

```diff
@@ -40,10 +41,30 @@ func (ca *Agent) Start(path string) error {
 	}
 	ca.c = c
 	go func() {
-		for range time.Tick(1 * time.Minute) {
+		var lastModTime time.Time
+		// Rarely, if two changes happen in the same second, mtime will
+		// be the same for the second change, and an mtime-based check would
+		// fail. Reload periodically just in case.
+		skips := 0
+		for range time.Tick(1 * time.Second) {
+			if skips < 600 {
+				// Check if the file changed to see if it needs to be re-read.
+				// os.Stat follows symbolic links, which is how ConfigMaps work.
+				stat, err := os.Stat(path)
+				if err != nil {
+					logrus.WithField("path", path).WithError(err).Error("Error loading config.")
+					continue
+				}
+				if stat.ModTime() == lastModTime {
+					skips++
+					continue // file hasn't been modified
+				}
+				lastModTime = stat.ModTime()
```

Comment:
We probably should only update the `lastModTime` after `Load` succeeds.
Currently, if `Load` fails after we update the lastModTime we won't try to reload again until the config is updated again or the skip limit is reached.

Reply:
er why would we try to keep reloading if it fails to load? that's pretty much what updating the config or the skip limit are for?

Reply:
This skip limit will let it recover eventually, I'm just thinking of errors besides invalid configs like failing to open the file. For those kind of error it would be best to retry again.
You're right though, this isn't really an issue, especially since those types of errors are not common.

Reply:
I think the only case where open fails when stat didn't is when you are out of FDs. I think we're fine here since we have the skip limit, which we can also tune.

Reply:
The most likely reason for Load to fail is a yaml syntax error, in which case we probably don't want to spam the log with errors every second.